event_routing_backends/processors/xapi/event_transformers/enrollment_events.py

```diff
@@ -26,17 +23,20 @@ def get_object(self):
         Returns:
             `Activity`
         """
-        course_id = self.event['context']['course_id']
+        course_id = self.get_data('context.course_id')
         object_id = make_course_url(course_id)
 
-        course = get_course_from_id(course_id)
-        display_name = course['display_name']
+        if course_id is not None:
+            course = get_course_from_id(course_id)
+            display_name = course['display_name']
+        else:
+            display_name = None
 
         return Activity(
             id=object_id,
             definition=ActivityDefinition(
                 type=constants.XAPI_ACTIVITY_COURSE,
-                name=LanguageMap({constants.EN: display_name}),
+                name=LanguageMap(**({constants.EN: display_name} if display_name is not None else {})),
```

Comment:
why we need None check here if we are striping None keys after transformation?

Reply:
This fields is of string type. So we can not assign None value. So I have implemented this check
TypeError: Value must be a stringstring_types